mantidproject/mantid
# BATS: normalise to monitor in wavelength

**Description of work.**
Changes the normalisation to monitor to be in wavelength instead of Energy.
Energy is not correct due to the issue in ConvertUnits: https://github.com/mantidproject/mantid/issues/32231 
Also adds a possibility to preserve the monitor spectra in the units of deltaE for diagnostic purposes.
System tests are slightly improved and tolerances are tightened.
Release note added.



**To test:**

1. In Settings>Plot, set normalize to bin width to none
2. Set x-axis scale to linear, y-axis scale to log
3. Run this script (the file is attached, remove the `.zip`, it's a nexus file) 
[318178.nxs.zip](https://github.com/mantidproject/mantid/files/7007906/318178.nxs.zip)

```
IndirectILLEnergyTransfer(
    OutputWorkspace='ST165_75RH_360K_lr4',
    Run='%YOUR_PATH_TO_RUN%/318178.nxs', 
    MonitorCutoff=0,
    SpectrumAxis='Q',
    InputElasticChannelWorkspace=None,
    OutputElasticChannelWorkspace='epp',
    ElasticPeakFitting='FitEquatorialOnly',
    NormaliseTo='Monitor',
    GroupDetectors=True,
    DiscardSingleDetectors=True,
    DeleteMonitorWorkspace=False
)
SumSpectra('ST165_75RH_360K_lr4', OutputWorkspace='PSDsum')
Scale(InputWorkspace='ST165_75RH_360K_lr4_318178_mon', OutputWorkspace='monscaled', Factor=3.3*1e-8)
```
4. Plot monscaled, then overplot PSDsum_1.
5. Zoom in to x-interval [-0.2, 0.25]
6. Observe that the frame covered by the monitor covers more-or-less the frame of the detector data, unlike in the attachment (note that the plot is in E and not deltaE)
![BATS-monitor-Mantid](https://user-images.githubusercontent.com/16865759/129914425-625f00d8-2835-4566-9e4c-e8bb557c0d3c.png)



Fixes #32230 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
